pkg/resolver/related.go

```diff
@@ -262,88 +262,69 @@ func (s *SearchResult) getRelations(ctx context.Context) []SearchRelatedResult {
 		s.updateKindMap(uid, kind, relatedMap) // Store result in a map
 	}
 
-	// retrieve provided kind filters only
-	if len(s.input.RelatedKinds) > 0 {
-		s.relatedKindUIDs(relatedMap) // get uids for provided kinds
-		// Build Related kinds query
-		s.buildRelatedKindsQuery()
-		items, err := s.resolveItems() // Fetch the related kind items
-		if err != nil {
-			klog.Warning("Error resolving relatedKind items", err)
-		} else { // Convert to (kind, items) format - when relatedKinds are requested
-			relatedSearch = s.searchRelatedResultKindItems(items)
-		}
-	} else { // Retrieve kind and count of related items
-		relatedSearch = s.searchRelatedResultKindCount(relatedMap)
+	// get uids for related items that match the relatedKind filter.
+	s.relatedKindUIDs(relatedMap)
+
+	// Build Related kinds query
+	s.buildRelatedKindsQuery()
+	items, err := s.resolveItems() // Fetch the related items
+	if err != nil {
+		klog.Warning("Error resolving related items.", err)
+		return []SearchRelatedResult{}
 	}
-	klog.V(5).Info("relatedSearch: ", relatedSearch)
+
+	// Convert to (kind, count, items) format
+	relatedSearch = s.searchRelatedResultKindItems(items)
+
+	klog.V(5).Info("RelatedSearch Result: ", relatedSearch)
 	return relatedSearch
 }
 
+// Gets the UIDs of relationships that match the relatedKinds filter.
 func (s *SearchResult) relatedKindUIDs(levelsMap map[string][]string) {
 	klog.V(6).Info("levelsMap in relatedKindUIDs: ", levelsMap)
 
-	relatedKinds := pointerToStringArray(s.input.RelatedKinds)
 	s.uids = []*string{}
-	keys := getKeys(levelsMap)
-	for _, kind := range relatedKinds {
-		// Convert kind to right case even if incoming query in RelatedKinds is all lowercase
-		// Needed for V1 compatibility.
-		for _, key := range keys {
-			if strings.EqualFold(key, kind) {
-				s.uids = append(s.uids, stringArrayToPointer(levelsMap[key])...)
-				break
+
+	// If relatedKinds filter is empty, need to resolve all.
+	if len(s.input.RelatedKinds) == 0 {
+		for _, values := range levelsMap {
+			s.uids = append(s.uids, stringArrayToPointer(values)...)
+		}
+	} else {
+		// Get UIDs of related items that match the kind filter.
+		kinds := getKeys(levelsMap)
+		for _, kindFilter := range s.input.RelatedKinds {
+			for _, kind := range kinds {
+				if strings.EqualFold(kind, *kindFilter) {
+					s.uids = append(s.uids, stringArrayToPointer(levelsMap[kind])...)
```

Comment:
Insert `break` here? If we get the kind, we are looking for, there is no need to look at the rest of the values.

Reply:
done